src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/matcher/CaseAccessGroupsMatcher.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.matcher;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ObjectUtils;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Component
+public class CaseAccessGroupsMatcher implements RoleAttributeMatcher {
+
+    @Override
+    public MatcherType getType() {
+        return MatcherType.ROLENAME;
+    }
+
+    private static final String VALUE_FIELD = "value";
+    private static final String ID_FIELD = "id";
+    private static final String CASE_ACCESS_GROUP_ID_FIELD = "caseAccessGroupId";
+
+    private static final Logger LOG = LoggerFactory.getLogger(CaseAccessGroupsMatcher.class);
+
+    @Override
+    public boolean matchAttribute(RoleAssignment roleAssignment, CaseDetails caseDetails) {
+        Optional<String> raCaseAccessGroupId = roleAssignment.getAttributes().getCaseAccessGroupId();
+        List<String> caseAccessGroupIds = getCaseAccessGroupIds(caseDetails).orElse(Collections.emptyList());
+        LOG.info("Match role assignment caseAccessGroupId {} with caseAccessGroupIds {} for role assignment {}",
+            raCaseAccessGroupId,
+            caseAccessGroupIds,
+            roleAssignment.getId());
+        boolean matched = caseAccessGroupIds.isEmpty() && (ObjectUtils.isEmpty(raCaseAccessGroupId));
+
+        for (String caseAccessGroupId : caseAccessGroupIds) {
+            if (isValuesMatching(raCaseAccessGroupId, caseAccessGroupId)) {
+                matched = true;
+                break;
+            }
+        }
+
+        log.info("Role assignment caseAccessGroupId {} and case details caseAccessGroupIds {} match {}",
```

Comment:
```suggestion
        log.debug("Role assignment caseAccessGroupId {} and case details caseAccessGroupIds {} match {}",
```
same comment as above - `debug` level